src/datasource.ts

```diff
@@ -6,34 +6,41 @@ import {
   MutableDataFrame,
   FieldType,
   TimeRange,
+  DataSourceWithLogsVolumeSupport,
+  LoadingState,
 } from '@grafana/data';
 import { getBackendSrv, getTemplateSrv } from '@grafana/runtime';
-import { lastValueFrom , Observable , Subject , of , from , map } from 'rxjs';
+import { lastValueFrom , Observable , Subject , of , from , map , startWith  } from 'rxjs';
 import { calculateInterval, createEpochTimeRangeBoundary } from './utils/time.utils';
 import { SumoApiDimensionValuesResponse } from './types/metadataCatalogApi.types';
 import { mapKeyedErrorMessage } from './utils/keyedErrors.utils';
 import { MetricsQuery, SumoApiMetricsResponse, SumoQuery } from './types/metricsApi.types';
 import { dimensionsToLabelSuffix } from './utils/labels.utils';
 import { interpolateVariable } from './utils/interpolation.utils';
 
-import { formatSumoValues, getSumoToGrafanaType, SumoQueryType } from './types/constants';
+import { isLogsQuery, SumoQueryType } from './types/constants';
 import { searchQueryExecutionService } from './services/logDataService/searchQueryExecutionService';
 import { MAX_NUMBER_OF_RECORDS, RunSearchActionType, SearchQueryType, SearchStatus } from './services/logDataService/constants';
-import { IField, ISearchStatus, RunSearchAction, SearchQueryParams } from './services/logDataService/types';
+import { ISearchStatus, RunSearchAction, SearchQueryParams, SearchResult } from './services/logDataService/types';
 
-import { filter , switchMap } from 'rxjs/operators';
+import { generateAggregateResponse, generateFullRangeHistogram, generateNonAggregateResponse, IPreviousCount, searchFilterForAggregateQuery, searchFilterForNonAggregateQuery } from './services/logDataService/logSearch.utils';
 
-export class DataSource extends DataSourceApi<SumoQuery> {
+export class DataSource extends DataSourceApi<SumoQuery> 
+implements DataSourceWithLogsVolumeSupport<SumoQuery>
```

Comment:
this one is removed in the newest version already https://github.com/grafana/grafana/pull/66266

Reply:
please use this one instead `DataSourceWithSupplementaryQueriesSupport` and replace `getLogsVolumeDataProvider` with appropriate method

Reply:
Also they mentioned the API is for internal use only. Not sure if we should use it. Is there any public alternative available?

Reply:
No im not able to find any public method for this one.

Reply:
@akononenko-sumo `DataSourceWithSupplementaryQueriesSupport` is not available on grafana dependency version we are using in this repo.

Reply:
hm, I believe you can update then

Reply:
maybe they intentionally closed access to this? I mean we are using internal API here